integration/helpers.go

```diff
@@ -222,26 +221,38 @@ func AssertMatchesCommandError(t testing.TB, expected, actual error) {
 	}
 }
 
-// AssertMatchesWriteError asserts error codes are the same.
-//
-// TODO check not only code; make is look similar to AssertMatchesCommandError above.
-// https://github.com/FerretDB/FerretDB/issues/2545
+// AssertMatchesWriteError asserts that both errors are equal WriteErrors,
+// except messages (and ignoring the Raw part).
 func AssertMatchesWriteError(t testing.TB, expected, actual error) {
 	t.Helper()
 
-	var aErr, eErr mongo.WriteException
+	a, okA := actual.(mongo.WriteException)   //nolint:errorlint // do not inspect error chain
+	e, okE := expected.(mongo.WriteException) //nolint:errorlint // do not inspect error chain
 
-	if ok := errors.As(actual, &aErr); !ok || len(aErr.WriteErrors) != 1 {
+	if !okA && !okE {
+		// driver sent an error which is not WriteException, check errors are the same.
 		assert.Equal(t, expected, actual)
```

Comment:
That would happily allow `AssertMatchesWriteError(t, io.EOF, io.EOF)`, but that contradicts the function description

Reply:
FYI, currently I'm seeing error from the driver that _update needs to have at least one field and cannot be empty_. I guess we are testing that in few tests.

Reply:
Example https://github.com/FerretDB/FerretDB/actions/runs/4955317748/jobs/8864615428

Reply:
`AssertMatchesWriteError` should check that both errors are `WriteError`. It should not consider other error types to be acceptable. That line of code is not correct

Reply:
Updated to assert only for `WriteError`. For `ReplaceEmptyDocument` test, given the error comes from mongo go driver https://github.com/mongodb/mongo-go-driver/blob/master/mongo/mongo.go#L158, not sure if we want to test driver errors. But for now, I'm checking that specific error message for target and compat before calling `WriteError`. 
https://github.com/FerretDB/FerretDB/blob/bab38aeab4cf2919ee0b633d773a9c68698e2cfc/integration/update_compat_test.go#L435